src/Npgsql/TypeHandlers/CompositeHandlers/CompositeHandler.cs

```diff
@@ -97,31 +101,133 @@ public override int ValidateAndGetLength(T value, ref NpgsqlLengthCache? lengthC
             return lengthCache.Lengths[position] = length;
         }
 
-        CompositeMemberHandler<T>[] CreateMemberHandlers() => CreateMemberHandlers((PostgresCompositeType)PostgresType, _typeMapper, _nameTranslator);
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        void Initialize()
+        {
+            if (_memberHandlers is null)
+                InitializeCore();
 
-        static CompositeMemberHandler<T>[] CreateMemberHandlers(PostgresCompositeType pgType, ConnectorTypeMapper typeMapper, INpgsqlNameTranslator nameTranslator)
+            void InitializeCore()
+            {
+                var pgType = (PostgresCompositeType)PostgresType;
+
+                _memberHandlers = CreateMemberHandlers(pgType, _typeMapper, _nameTranslator);
+                _constructorHandler = CreateConstructorHandler(pgType, _typeMapper, _nameTranslator);
+                _constructor = _constructorHandler is null
+                    ? Expression
+                        .Lambda<Func<T>>(Expression.New(typeof(T)))
+                        .Compile()
+                    : null;
+            }
+        }
+
+        static CompositeConstructorHandler<T>? CreateConstructorHandler(PostgresCompositeType pgType, ConnectorTypeMapper typeMapper, INpgsqlNameTranslator nameTranslator)
         {
+            var pgFields = pgType.Fields;
+
             var clrType = typeof(T);
+            var clrParameterHandlerType = typeof(CompositeParameterHandler<>);
```

Comment:
Let's inline these for readability. I checked and there's no perf impact:

```c#
[Benchmark]
public int Inside()
{
    var sum = 0;
    for (var i = 0; i < 100; i++)
        sum += typeof(TypeLoading).Name.Length;
    return sum;
}

[Benchmark]
public int Outside()
{
    var sum = 0;
    var t = typeof(TypeLoading);
    for (var i = 0; i < 100; i++)
        sum += t.Name.Length;
    return sum;
}
```

Results:

```
BenchmarkDotNet=v0.11.5, OS=ubuntu 19.04
Intel Xeon W-2133 CPU 3.60GHz, 1 CPU, 12 logical and 6 physical cores
.NET Core SDK=3.0.100
  [Host]     : .NET Core 3.0.0 (CoreCLR 4.700.19.46205, CoreFX 4.700.19.46214), 64bit RyuJIT
  DefaultJob : .NET Core 3.0.0 (CoreCLR 4.700.19.46205, CoreFX 4.700.19.46214), 64bit RyuJIT


|  Method |     Mean |     Error |    StdDev |
|-------- |---------:|----------:|----------:|
|  Inside | 512.4 ns | 0.6466 ns | 0.6048 ns |
| Outside | 538.6 ns | 1.5697 ns | 1.4683 ns |
```

Reply:
Moved the variable inside the loop. Is it better?

Reply:
I think so :)